SFS_time_domain/driving_function_imp_nfchoa.m

```diff
@@ -129,24 +132,38 @@
     end
 end
 
-% -------------------------------------------------------------------------
-%                      ___
-%                1     \
-% D(phi0,w) = -------  /__     Hm(w) e^(im(phi0-phi_src))
-%             2 pi r0  m=-M..M
-%
-% see Spors et al. (2011), eq.(4) and (5)
-%--------------------------------------------------------------------------
+if strcmp('2.5D',dimension)
+    % -------------------------------------------------------------------------
+    %                      ___
+    %                1     \
+    % D(phi0,w) = -------  /__     Hm(w) e^(im(phi0-phi_src))
+    %             2 pi r0  m=-M..M
+    %
+    % see Spors et al. (2011), eq.(4) and (5)
+    %--------------------------------------------------------------------------
+    
+    % apply modal weighting
+    wm = modal_weighting(order,conf);  % modal weighting function
+    Hm = bsxfun(@times,Hm,wm);
```

Comment:
Is this only the case for the max-re window?
I also see no apparent alternative. At the moment we simply use no weighting window in the 3D case which you would not expect as a user if you have set the conf setting to use one window.
If you would not distinguish between 2D/3D directly in `modal_weighting` how would you then handle it? Introduce another `modal_weighting_3D` function, or is it no longer called modal weighting in the 3D case.